File: spec/javascript/test_ts_packet_header.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('TsPacketHeader', 'src/ts_packet.bin', function(r, TsPacketHeader) {
  assert.equal(r.syncByte, 0x47);
  assert.equal(r.transportErrorIndicator, false);
  assert.equal(r.payloadUnitStartIndicator, false);
  assert.equal(r.transportPriority, true);
  assert.equal(r.pid, 33);
  assert.equal(r.transportScramblingControl, 0);
  assert.equal(r.adaptationFieldControl, TsPacketHeader.AdaptationFieldControlEnum.PAYLOAD_ONLY);
});

File: spec/java/src/io/kaitai/struct/spec/TestTsPacketHeader.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.TsPacketHeader;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestTsPacketHeader extends CommonSpec {
    @Test
    public void testTsPacketHeader() throws Exception {
        TsPacketHeader r = TsPacketHeader.fromFile(SRC_DIR + "ts_packet.bin");

        assertEquals(r.syncByte(), 0x47);
        assertEquals(r.transportErrorIndicator(), false);
        assertEquals(r.payloadUnitStartIndicator(), false);
        assertEquals(r.transportPriority(), true);
        assertEquals(r.pid(), 33);
        assertEquals(r.transportScramblingControl(), 0);
        assertEquals(r.adaptationFieldControl(), TsPacketHeader.AdaptationFieldControlEnum.PAYLOAD_ONLY);
    }
}
